extractor/extractor/index.py

```diff
@@ -108,10 +140,26 @@ def __iter__(self) -> Iterator[IndexRecord]:
             if passed:
                 yield record
 
+    def augment(self, other: Iterable[IndexRecord]) -> Index:
+        return AugmentedIndex(self, other)
+
     def filter(self, cb: FilterCallback[IndexRecord]) -> Index:
         return FilteredIndex(self, cb)
 
 
+class AugmentedIndex(Index):
```

Comment:
!

Reply:
That may go away, I changed my mind about how some of that should work. The whole `Index` thing is gonna be refactored into a single base class with implementations for the BMF and main indexes.